Mathlib/Probability/Kernel/Disintegration/Density.lean

```diff
@@ -0,0 +1,799 @@
+  (measurable_densityProcess κ ν n hs).comp (measurable_const.prod_mk measurable_id)
+
+lemma measurable_countableFiltration_densityProcess (κ : kernel α (γ × β)) (ν : kernel α γ) (n : ℕ)
+    (a : α) {s : Set β} (hs : MeasurableSet s) :
+    Measurable[countableFiltration γ n] (fun x ↦ densityProcess κ ν n a x s) := by
+  refine @Measurable.ennreal_toReal _ (countableFiltration γ n) _ ?_
+  exact (measurable_densityProcess_countableFiltration_aux κ ν n hs).comp measurable_prod_mk_left
+
+lemma stronglyMeasurable_countableFiltration_densityProcess (κ : kernel α (γ × β)) (ν : kernel α γ)
+    (n : ℕ) (a : α) {s : Set β} (hs : MeasurableSet s) :
+    StronglyMeasurable[countableFiltration γ n] (fun x ↦ densityProcess κ ν n a x s) :=
+  (measurable_countableFiltration_densityProcess κ ν n a hs).stronglyMeasurable
+
+lemma adapted_densityProcess (κ : kernel α (γ × β)) (ν : kernel α γ) (a : α)
+    {s : Set β} (hs : MeasurableSet s) :
+    Adapted (countableFiltration γ) (fun n x ↦ densityProcess κ ν n a x s) :=
+  fun n ↦ stronglyMeasurable_countableFiltration_densityProcess κ ν n a hs
+
+lemma densityProcess_nonneg (κ : kernel α (γ × β)) (ν : kernel α γ) (n : ℕ)
+    (a : α) (x : γ) (s : Set β) :
+    0 ≤ densityProcess κ ν n a x s :=
+  ENNReal.toReal_nonneg
+
+lemma meas_countablePartitionSet_le_of_fst_le (hκν : fst κ ≤ ν) (n : ℕ) (a : α) (x : γ)
+    (s : Set β) :
+    κ a (countablePartitionSet n x ×ˢ s) ≤ ν a (countablePartitionSet n x) := by
+  calc κ a (countablePartitionSet n x ×ˢ s)
+    ≤ fst κ a (countablePartitionSet n x) := by
+        rw [fst_apply' _ _ (measurableSet_countablePartitionSet _ _)]
+        refine measure_mono (fun x ↦ ?_)
+        simp only [mem_prod, mem_setOf_eq, and_imp]
+        exact fun h _ ↦ h
+  _ ≤ ν a (countablePartitionSet n x) := hκν a _
+
+lemma densityProcess_le_one (hκν : fst κ ≤ ν) (n : ℕ) (a : α) (x : γ) (s : Set β) :
+    densityProcess κ ν n a x s ≤ 1 := by
+  refine ENNReal.toReal_le_of_le_ofReal zero_le_one (ENNReal.div_le_of_le_mul ?_)
+  rw [ENNReal.ofReal_one, one_mul]
+  exact meas_countablePartitionSet_le_of_fst_le hκν n a x s
+
+lemma snorm_densityProcess_le (hκν : fst κ ≤ ν) (n : ℕ) (a : α) (s : Set β) :
+    snorm (fun x ↦ densityProcess κ ν n a x s) 1 (ν a) ≤ ν a univ := by
+  refine (snorm_le_of_ae_bound (C := 1) (ae_of_all _ (fun x ↦ ?_))).trans ?_
+  · simp only [Real.norm_eq_abs, abs_of_nonneg (densityProcess_nonneg κ ν n a x s),
+      densityProcess_le_one hκν n a x s]
+  · simp
+
+lemma integrable_densityProcess (hκν : fst κ ≤ ν) [IsFiniteKernel ν] (n : ℕ)
+    (a : α) {s : Set β} (hs : MeasurableSet s) :
+    Integrable (fun x ↦ densityProcess κ ν n a x s) (ν a) := by
+  rw [← memℒp_one_iff_integrable]
+  refine ⟨Measurable.aestronglyMeasurable ?_, ?_⟩
+  · exact measurable_densityProcess_right κ ν n a hs
+  · exact (snorm_densityProcess_le hκν n a s).trans_lt (measure_lt_top _ _)
+
+lemma set_integral_densityProcess_I (hκν : fst κ ≤ ν) [IsFiniteKernel κ] [IsFiniteKernel ν]
```

Comment:
I don't understand the naming `_I`. Also, doesn't the finiteness assumption on kappa follow from `hκν` (here and below)?